app/models/name/synonymy.rb

```diff
@@ -47,25 +47,21 @@ def sort_synonyms
 
   # Same as +other_authors+, but returns ids.
   def other_author_ids
-    @other_author_ids ||= begin
-      if @other_authors
-        @other_authors.map(&:id)
-      else
-        Name.pluck(:id).where(text_name: text_name).map(&:to_i)
-      end
-    end
```

Comment:
Question: **Is the `begin ... end` block necessary?**
(In removing it, I might have missed something basic/subtle/deep.)